pkgs/by-name/gd/gdlauncher-carbon/package.nix

```diff
@@ -0,0 +1,119 @@
+{ lib
+, stdenv
+, appimageTools
+, fetchurl
+, graphicsmagick
+, makeWrapper
+, copyDesktopItems
+, autoPatchelfHook
+, xorg
+, libpulseaudio
+, libGL
+, udev
+, xdg-utils
+, electron
+, addOpenGLRunpath
+, makeDesktopItem
+
+, jdk8
+, jdk17
+, jdks ? [ jdk17 jdk8 ]
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "gdlauncher-carbon";
+  version = "2.0.2";
+
+  src = appimageTools.extract {
+    inherit (finalAttrs) pname version;
+    src = fetchurl {
+      url = "https://cdn-raw.gdl.gg/launcher/GDLauncher__${finalAttrs.version}__linux__x64.AppImage";
+      hash = "sha256-mPvQK+4s0BxR8yi0Fu6HMtWuH7bKj4jvTBFVa+cbZs8=";
+    };
+  };
+
+  nativeBuildInputs = [
+    graphicsmagick
+    makeWrapper
+    copyDesktopItems
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    xorg.libxcb
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/gdlauncher-carbon/resources
+    cp -r $src/resources/{binaries,app.asar} $out/share/gdlauncher-carbon/resources/
+
+    # The provided icon is a bit large for some systems, so make smaller ones
```

Comment:
I think this is done automatically by all relevant icon users?

Reply:
Perhaps I can try again; last time I checked, it didn't seem to work on KDE Plasma.